tests/UI/campaigns/productV2/sanity/03_CRUDProductWithCombinations.js

```diff
@@ -0,0 +1,278 @@
+      await expect(successMessage).to.equal('Successfully generated 4 combinations.');
+    });
+
+    it('should close generate combinations modal', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'closeGenerateCombinationsModal', baseContext);
+
+      const isModalVisible = await createProductsPage.closeGenerateCombinationModal(page);
+      await expect(isModalVisible).to.be.true;
+    });
+
+    it('should save the product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'saveProduct', baseContext);
+
+      const updateProductMessage = await createProductsPage.saveProduct(page);
+      await expect(updateProductMessage).to.equal(createProductsPage.successfulUpdateMessage);
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct', baseContext);
+
+      // Click on preview button
+      page = await createProductsPage.previewProduct(page);
+
+      await foProductPage.changeLanguage(page, 'en');
+
+      const pageTitle = await foProductPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(newProductData.name);
+    });
+
+    it('should check all product information', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkProductInformation', baseContext);
+
+      let result = await foProductPage.getProductInformation(page);
+      await Promise.all([
+        await expect(result.name).to.equal(newProductData.name),
+        await expect(result.price).to.equal(newProductData.price),
+        await expect(result.shortDescription).to.equal(newProductData.summary),
+        await expect(result.description).to.equal(newProductData.description),
+      ]);
+
+      result = await foProductPage.getProductAttributes(page);
+      await Promise.all([
+        await expect(result.size).to.equal(newProductData.combinations.size.join(' ')),
+        await expect(result.color).to.equal(newProductData.combinations.color.join(' ')),
+      ]);
+    });
+
+    it('should go back to BO', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goBackToBO', baseContext);
+
+      // Go back to BO
+      page = await foProductPage.closePage(browserContext, page, 0);
+
+      const pageTitle = await createProductsPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(createProductsPage.pageTitle);
+    });
+  });
+
+  describe('Edit product', async () => {
+    it('should edit the created product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'editProduct', baseContext);
+
+      const createProductMessage = await createProductsPage.setProduct(page, editProductData);
+      await expect(createProductMessage).to.equal(createProductsPage.successfulUpdateMessage);
+    });
+
+    it('should add combinations and check generate combinations button', async function () {
```

Comment:
I am not sure if I understand it right, but based on what @jolelievre said. Maybe here would be more accurate with `should add attributes and check generate combinations button`. I may be mistaking :)